model.py

```diff
@@ -6969,8 +6969,9 @@ def extension(self, destination_type=None):
         """Try to come up with a good file extension for this representation."""
         if destination_type:
             return self._extension(destination_type)
-        else:
-            return self.url_extension or self._extension(self.clean_media_type)
+        return self._extension(
+            self._clean_media_type(self.external_media_type)
+        ) or self.url_extension
```

Comment:
If I don't switch the order of these two, then the url_extension for foo.svg returns ".svg" and we don't get a chance to override that result.